### PR TITLE
feat(optimizer): Improve cardinality estimation for join based on histogram

### LINKED ISSUE #9409: Improve cardinality estimation for join operator

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #9409

### CODE REVIEW

**Comment by BohuTANG** [comment]
It seems `test_sqllogic_ydb_cluster_linux` always timeout. Try to reduce the parallel threads?

**Comment by Xuanwo** [comment]
> It seems `test_sqllogic_ydb_cluster_linux` always timeout. Try to reduce the parallel threads?

Some thing must be buggy. We have a 30m timeout, and `test_sqllogic_ydb_standalone_linux` finished in 11m, `test_sqllogic_ydb_cluster_linux` should not be timeout.

**Comment by xudong963** [comment]
> It seems `test_sqllogic_ydb_cluster_linux` always timeout. Try to reduce the parallel threads?

I'm debuging it in https://github.com/datafuselabs/databend/pull/9595

**Comment by BohuTANG** [comment]
@mergify update

**Comment by xudong963** [comment]
> > It seems `test_sqllogic_ydb_cluster_linux` always timeout. Try to reduce the parallel threads?
> 
> I'm debuging it in #9595

I just enable `--debug` arg, and it seems ci becomes smooth 🤣 

image

Continuous observation is required

**Comment by BohuTANG** [comment]
https://github.com/datafuselabs/databend/pull/9615 fixed the timeout of logictest.

**Comment by xudong963** [comment]
@mergify update

File: src/query/sql/src/planner/optimizer/property/column_stat.rs [hunk 1]
```diff
@@ -15,13 +15,23 @@
...
 /// Statistics information of a column
 pub struct ColumnStat {
+    /// Min value of the column
+    pub min: Datum,
+
+    /// Max value of the column
+    pub max: Datum,
+
+    /// Number of distinct values
+    pub ndv: u64,
```
Comment (leiysky) [inline src/query/sql/src/planner/optimizer/property/column_stat.rs:33]: Better to use `f64` here.

**Review by leiysky — APPROVED** [review]
Rest LGTM

File: src/query/sql/src/planner/optimizer/property/column_stat.rs (hunk 1 above)
Comment (xudong963) [reply src/query/sql/src/planner/optimizer/property/column_stat.rs:33]: Will do it in next PR.

**Pull request merged**